src/android/toga_android/widgets/numberinput.py

```diff
@@ -13,17 +13,29 @@
 from .base import Widget
 
 
-class TogaTextWatcher(TextWatcher):
-    def __init__(self, impl):
+class TogaNumberInputWatcher(TextWatcher):
+    def __init__(self, number_input_interface):
         super().__init__()
-        self._impl = impl
+        self.interface = number_input_interface
 
     def beforeTextChanged(self, _charSequence, _i, _i1, _i2):
         pass
 
     def afterTextChanged(self, _editable):
-        if self._impl.interface.on_change:
-            self._impl.interface.on_change(widget=self._impl.interface)
+        current_string = self.interface._impl.native.getText().toString()
```

Comment:
Is this round trip needed, instead of `self.native` directly?

Reply:
I think `self.native` is fine, and I'll go for that and fix this PR.

Reply:
You fixed this -- thanks! Resolving.